tests still failing on compile SDK 28

Even with #7870 and #7871 included tests fail after upgrading _compile_ SDK to 28, see https://ci.cgeo.org/job/cgeo%20pull%20request/2325/consoleFull#-109812861545fc375f-a63e-468b-8450-76255356fdf9:

```
> Task :main:compileBasicDebugAndroidTestJavaWithJavac FAILED
16:51:04  /home/jenkins/slave/workspace/cgeo pull request/tests/src-android/cgeo/CGeoTestCase.java:9: error: package android.test does not exist
16:51:04  import android.test.ApplicationTestCase;
16:51:04                     ^
16:51:04  /home/jenkins/slave/workspace/cgeo pull request/tests/src-android/cgeo/CGeoTestCase.java:13: error: cannot find symbol
16:51:04  public abstract class CGeoTestCase extends ApplicationTestCase<CgeoApplication> {
16:51:04                                             ^
16:51:04    symbol: class ApplicationTestCase
16:51:04  /home/jenkins/slave/workspace/cgeo pull request/tests/src-android/cgeo/geocaching/test/AbstractResourceInstrumentationTestCase.java:20: error: package android.test does not exist
16:51:04  import android.test.InstrumentationTestCase;
16:51:04                     ^
16:51:04  /home/jenkins/slave/workspace/cgeo pull request/tests/src-android/cgeo/geocaching/test/AbstractResourceInstrumentationTestCase.java:32: error: cannot find symbol
16:51:04  public abstract class AbstractResourceInstrumentationTestCase extends InstrumentationTestCase {
16:51:04                                                                        ^
16:51:04    symbol: class InstrumentationTestCase
16:51:04  /home/jenkins/slave/workspace/cgeo pull request/tests/src-android/cgeo/geocaching/connector/gc/GCParserTest.java:28: error: package android.test.suitebuilder.annotation does not exist
...
```

@bekuno  Can you have a look at it?

--- a/build.gradle
+++ b/build.gradle
@@ -155,7 +155,7 @@ buildScan {
  */
 
 ext {
-    supportLibraryVersion = '27.1.1'
+    supportLibraryVersion = '28.0.0'
 }
 
 subprojects {

--- a/cgeo-contacts/build.gradle
+++ b/cgeo-contacts/build.gradle
@@ -5,7 +5,7 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 27
+    compileSdkVersion 28
 
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8

--- a/main/build.gradle
+++ b/main/build.gradle
@@ -18,7 +18,7 @@ if (isContinuousIntegrationServer()) {
 }
 
 android {
-    compileSdkVersion 27
+    compileSdkVersion 28
 
     buildToolsVersion "28.0.3"
 
@@ -239,6 +239,10 @@ android {
         }
         // debug and release builds have offset zero, no special handling here
     }
+
+    // tests rely on JUnit-based classes
+    useLibrary 'android.test.runner'
+    useLibrary 'android.test.base'
 }
 
 dependencies {

--- a/mapswithme-api/build.gradle
+++ b/mapswithme-api/build.gradle
@@ -3,7 +3,7 @@ apply plugin: 'android-library'
 android {
 
   // Define these properties in the gradle.properties file in the root project folder
-  compileSdkVersion 27
+  compileSdkVersion 28
 
   defaultConfig {
     minSdkVersion 16

